packages/pages-shared/src/asset-server/rulesEngine.ts

```diff
@@ -67,11 +83,17 @@ export const generateRulesMatcher = <T>(
 	][];
 
 	return ({ request }: { request: Request }) => {
-		const { pathname, host } = new URL(request.url);
+		const { pathname, host, search } = new URL(request.url);
+		const sortedSearch =
+			search && search.length
+				? "?" + search.slice(1).split("&").sort().join("&")
```

Comment:
I'd rather use the built-in `searchParams` for parsing.

```suggestion
				? "?" + new URLSearchParams([...searchParams.entries()].sort(([a], [b]) => a.localeCompare(b))).toString()
```